### PR TITLE
Don't time out WFT in case of subsequent UNHANDLED_COMMAND failures

### PR DESCRIPTION
**Pull request by Spikhalskiy**
Facing some edge cases with timing out WFTs on subsequent UNHANDLED_COMMAND, Temporal Server will stop doing the new behavior for UNHANDLED_COMMAND in 1.17
Incorporate the new change in Java Test Server early to mitigate facing these edge cases in stress tests

### CODE REVIEW

File: temporal-test-server/src/main/java/io/temporal/internal/testservice/TestWorkflowMutableStateImpl.java [hunk 1]
```diff
@@ -1007,16 +1011,20 @@ private StateMachine<SignalExternalData> getSignal(String signalId) {
...
   }
 
   private void processFailWorkflowTask(
-      RespondWorkflowTaskFailedRequest request, RequestContext ctx) {
+      RespondWorkflowTaskFailedRequest request,
+      RequestContext ctx,
+      boolean timeoutWorkflowTaskIfReoccurringFailure) {
     WorkflowTaskData data = workflowTaskStateMachine.getData();
-    if (data.attempt >= 2) {
+    if (timeoutWorkflowTaskIfReoccurringFailure && data.attempt >= 2) {
```
Comment (bergundy) [inline temporal-test-server/src/main/java/io/temporal/internal/testservice/TestWorkflowMutableStateImpl.java:1027]:
```suggestion
    if (timeoutWorkflowTaskIfRecurringFailure && data.attempt >= 2) {
```

**Review by bergundy — APPROVED** [review]
Just one typo
